### ISSUE TITLE
Add controller Support

### ISSUE THREAD

**Issue by LightningMods** [GracesGames/SimpleFileBrowser#38]
### Question, bug or feature

Im requesting that this browser get PS4 Controller Support 

### Version(s) affected

Latest

### Unity version

2.0p2
### Screenshot(s)
None

### Steps to reproduce

Add controller support

**Comment by LightningMods** [comment]
Yes, It works on the PS4 OS, it would be great if you could add the up and down dpad as the selector and X button to select, Also O button for back, i have attached a PS4 Button layout from the PS4 Unity Guide for you Thanks Alot!
![dualshock4inputmap](https://user-images.githubusercontent.com/34066913/38562908-a8d5bc90-3ca9-11e8-9651-a070415c349b.png)

**Comment by LightningMods** [comment]
Here is a screenshot of the File Manager working on PS4 :)
![ps4](https://user-images.githubusercontent.com/34066913/38563111-1b7d52e4-3caa-11e8-988f-0377c5c90c0c.png)

**Comment by GracesGames** [comment]
The screenshot you posted earlier was from:

[C# file browser](https://assetstore.unity.com/packages/tools/c-file-browser-18308)

While this is the GitHub repo of:

[Simple File Browser](https://assetstore.unity.com/packages/tools/c-file-browser-18308)

Are we talking about the asset? 😅 
If not, feel free to test mine on the PS4 as well, as it does support Linux ;)

**Comment by LightningMods** [comment]
Yes we are.. just i posted the wrong pick i got the file browsers mixed up, but i did test yours on PS4 but i cant get past this screen (see the pic) because the controls for PS4 are not configured etc and the controller doesn't
![screenshot_1](https://user-images.githubusercontent.com/34066913/38606273-90338894-3d43-11e8-9fea-9dcc0cf2abbf.png)
 do anything because of that i have included a PS4 Button layout in the previous post so you can add that feature thnx :)

**Comment by TheRealCatherine** [comment]
I would also love controller support but for XBox controllers. The hard thing, though, is providing cross platform controller support without bringing in external libraries.

My suggestion would be adding an api that simply has things like "GameObject NextItem(){}" and "GameObject HighlightedItem() {}" and making the user hook those up to their existing controller code.

**Comment by GracesGames** [comment]
That is indeed the issue I've been struggling with, finding a solid solution could easily result in a complete rewrite using Unity Events or something for everything ;)

**Comment by TheRealCatherine** [comment]
My vote would be to have like a built in item highlighting feature for selecting files/directories

![image](https://user-images.githubusercontent.com/31754592/43427690-5c6ead98-9420-11e8-9301-e6fc9b6a0fc3.png)

 then simply like:

public void CursorUp(), CursorDown(), CursorLeft(), CursorRight(). Game developer could hook like D-pad into these directly.

if the cursor is over a file it would call FileClick automatically then the developer could hook, say, X or A to call SelectFile indirectly

if the cursor is over a directory nothing would be called automatically but if the users selects with the same button as confirming file selection it would call DirectoryClick (and make sure _currentFile is cleared)

so there would need to be like a CursorSelect() method that the game developer hooks their selection button (like X or A or start or whatever) into. It would check _currentFile, if its set then call SelectFile. If not it would check if a directory is under the cursor and if so call DirectoryClick()

I say only use the cursor for selecting directories and files and make the developer assign things like directory up, back, forward, cancel by binding to the existing public API

Barring some other solutions we will eventually get around to implementing it more or less like I described but that's probably a few months away.
